trunk/pardot.php

```diff
@@ -56,3 +55,120 @@
+    wp_register_script(
+        'pardot-editor',
+        plugins_url( $index_js, __FILE__ ),
+        $script_asset['dependencies'],
+        $script_asset['version']
+    );
+
+    $editor_css = 'build/index.css';
+    wp_register_style(
+        'pardot-editor',
+        plugins_url( $editor_css, __FILE__ ),
+        array(),
+        filemtime( "$dir/$editor_css" )
+    );
+
+    $style_css = 'build/style-index.css';
+    wp_register_style(
+        'pardot',
+        plugins_url( $style_css, __FILE__ ),
+        array(),
+        filemtime( "$dir/$style_css" )
+    );
+
+    wp_localize_script( 'build/index.js', 'ajaxurl', admin_url( 'includes/admin-ajax.php' ));
+
+    register_block_type( 'pardot/form', array(
+        'editor_script'   => 'pardot-editor',
+        'editor_style'    => 'pardot-editor',
+        'style'           => 'pardot',
+        'render_callback' => 'pardot_form_block_callback',
+        'attributes'  => array(
+            'form_id'  => array(
+                'type'  => 'string',
+                'default' => '',
+            ),
+            'height'    => array(
+                'type'  => 'string',
+                'default'   => '',
+            ),
+            'width'    => array(
+                'type'  => 'string',
+                'default'   => '',
+            ),
+            'className'    => array(
+                'type'  => 'string',
+                'default'   => '',
+            ),
+            'title'    => array(
+                'type'  => 'string',
+                'default'   => '',
+            ),
+        ),
+    ) );
+
+    register_block_type( 'pardot/dynamic-content', array(
+        'editor_script'   => 'pardot-editor',
+        'editor_style'    => 'pardot-editor',
+        'style'           => 'pardot',
+        'render_callback' => 'pardot_dynamic_content_block_callback',
+        'attributes' => array(
+            'dynamicContent_id' => array(
+                'type' => 'string',
+                'default' => '',
+            ),
+            'dynamicContent_default' => array(
+                'type' => 'string',
+                'default' => '',
+            ),
+            'height' => array(
+                'type'  => 'string',
+                'default' => '',
+            ),
+            'width' => array(
+                'type'  => 'string',
+                'default' => '',
+            ),
+            'className' => array(
+                'type'  => 'string',
+                'default' => '',
+            ),
+        ),
+    ) );
+}
+
+
+add_action( 'init', 'pardot_init' );
+
+function pardot_form_block_callback($attributes) {
+    if (isset($attributes['form_id'])) {
+        $attributes['class'] = $attributes['className'];
+        unset($attributes['className']);
+        return Pardot_Plugin::get_form_body($attributes);
+    }
+    return '';
+}
+
+function pardot_dynamic_content_block_callback($attributes) {
+    if (isset($attributes['dynamicContent_id'])) {
```

Comment:
weird to see the mixed use of camel casing and underscores

Reply:
It's widely used in history code 🤷

Reply:
yeah, well, that's not correct though.

Reply:
Yeah it seems lowercase is what Wordpress encourages for variable and method names. https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#naming-conventions

Reply:
Our api does uses dynamicContent which makes sense to me why it uses dynamicContent vs dynamic_content instead. I don't see a quick way to find & replace existing all dynamicContent names in code. I tend to keep the current way.  https://developer.salesforce.com/docs/marketing/pardot/guide/dynamic-content-v3.html

Reply:
That's just an endpoint route, I don't think that's a good reason for keeping it. I am not a fan of it personally but if you think there's risk in replacing them with underscores, esp from this PR, then it's fine to keep things as is.